pkg/httpreverseproxy/client_test.go

```diff
@@ -0,0 +1,123 @@
+package httpreverseproxy
+
+import (
+	"net/http"
+	"net/http/httptest"
+	"net/http/httputil"
+	"net/url"
+	"testing"
+	"time"
+
+	libhttputil "github.com/nexmoinc/gosrvlib/pkg/httputil"
+	"github.com/nexmoinc/gosrvlib/pkg/testutil"
+	"github.com/stretchr/testify/require"
+)
+
+func TestNew(t *testing.T) {
+	t.Parallel()
+
+	tests := []struct {
+		name        string
+		serviceAddr string
+		opts        []Option
+		wantErr     bool
+	}{
+		{
+			name:        "fails with invalid character in URL",
+			serviceAddr: "http://invalid-url.domain.invalid\u007F",
+			wantErr:     true,
+		},
+		{
+			name:        "succeeds with defaults",
+			serviceAddr: "http://service.domain.invalid:1234/",
+			wantErr:     false,
+		},
+		{
+			name:        "succeeds with custom http client",
+			serviceAddr: "http://service.domain.invalid:1235/",
+			opts:        []Option{WithHTTPClient(&testHTTPClient{})},
+			wantErr:     false,
+		},
+		{
+			name:        "succeeds with custom reverse proxy",
+			serviceAddr: "http://service.domain.invalid:1236/",
+			opts:        []Option{WithReverseProxy(&httputil.ReverseProxy{})},
+			wantErr:     false,
+		},
+	}
+
+	for _, tt := range tests {
+		tt := tt
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+
+			c, err := New(tt.serviceAddr, tt.opts...)
+			if tt.wantErr {
+				require.Nil(t, c, "New() returned client should be nil")
+				require.Error(t, err, "New() error = %v, wantErr %v", err, tt.wantErr)
+				return
+			}
+
+			require.NotNil(t, c, "New() returned client should not be nil")
+			require.NoError(t, err, "New() unexpected error = %v", err)
+		})
+	}
+}
+
+func TestClient_ForwardRequest(t *testing.T) {
```

Comment:
If my other suggestion gets applied, this test would required adjustments to behave similarly to: https://github.com/nexmoinc/numapi/blob/e3cde95346bdcd387536b0f1346b3ac1ac78fd6c/internal/phub/phub_test.go#L131-L184